### PR TITLE
Update String documentation

### PR DESCRIPTION
**Pull request by surma**
UTF-16 is basically UCS-2 but with surrogate pairs. Since you are mentioning surrogate pairs here, it seems simpler to say that AssemblyScript stores strings in UTF-16 and not mention UCS-2 at all (but still link to @mathiasbynens’s article).

### CODE REVIEW

**Comment by dcodeIO** [comment]
Agree, the UCS-2 bit is unnecessary there. I believe I initially wanted to convey that much of the String API does not really care about surrogate pairs, simply representing these as subsequent code units, without going into the details on "potentially ill-formed UTF-16 or [WTF-16](https://simonsapin.github.io/wtf-8/)" too much there. Hard to make a statement that is both correct and can be understood. Perhaps if we say "potentially ill-formed UTF-16, just like JavaScript" there?

**Comment by mathiasbynens** [comment]
It's not quite UTF-16 because of lone surrogates. I like the suggestion to phrase it as "potentially ill-formed UTF-16, just like JavaScript" and link somewhere that explains the details.

File: standard-library/string.md [hunk 1]
```diff
@@ -71,5 +71,5 @@ The following mostly exist to have a safe way to copy between Strings and ArrayB
...
 
-Like JavaScript, AssemblyScript stores strings in [UTF-16 encoding represented by the API as UCS-2](https://mathiasbynens.be/notes/javascript-encoding), where certain UTF-8 **code points** are represented by two UTF-16 **code units**, so called surrogate pairs. This is done to mimic JavaScript as closely as possible, and in an attempt that calling JavaScript APIs from WebAssembly does not have to convert between encodings constantly. Whether or when future WebAssembly specifications will help in this regard is still [subject to discussion](https://github.com/WebAssembly/webidl-bindings/issues/13).
+AssemblyScript stores strings encoded as [UTF-16, just like JavaScript](https://mathiasbynens.be/notes/javascript-encoding), where certain Unicode **code points** are represented by two UTF-16 **code units**, so called surrogate pairs. This is done to mimic JavaScript as closely as possible, and in an attempt that calling JavaScript APIs from WebAssembly does not have to convert between encodings constantly. Whether or when future WebAssembly specifications will help in this regard is still [subject to discussion](https://github.com/WebAssembly/webidl-bindings/issues/13).
```
Comment (surma) [inline standard-library/string.md:74]:
```suggestion
AssemblyScript stores strings [the same way JavaScript does](https://mathiasbynens.be/notes/javascript-encoding), i.e. by encoding them using UTF-16. In UTF-16, certain Unicode **code points** are represented by two UTF-16 **code units** called surrogate pairs. However, just like in JavaScript, strings are not sanitized so lone surrogates can appear in a string, which are technically ill-formed UTF-16. This is done to mimic JavaScript as closely as possible, and avoids string re-encoding when calling JavaScript APIs from WebAssembly. Whether future WebAssembly specifications will help in this regard is still [subject to discussion](https://github.com/WebAssembly/webidl-bindings/issues/13).
```

**Comment by surma** [comment]
I gave it another spin. WDYT @dcodeIO @mathiasbynens ?

**Comment by mathiasbynens** [comment]
SGTM, thanks for doing this @surma!

**Comment by dcodeIO** [comment]
SGTM as well, thanks!

**Comment by surma** [comment]
Suggestion merged! Thanks for the review :)

**Pull request merged**